File: services/identity-service/src/test/java/br/com/deltaglobalbank/identity/domain/user/ApiClientTests.java
package br.com.deltaglobalbank.identity.domain.user;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;

import java.util.UUID;

import br.com.deltaglobalbank.identity.domain.apiClient.ApiClient;
import br.com.deltaglobalbank.identity.domain.apiClient.ApiClientStatus;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class ApiClientTests {

    @Test
    void mustInitializeWithActiveStatus() {
        ApiClient apiClient = ApiClient.newApiClient(UUID.randomUUID(), UUID.randomUUID(), "client", null);
        assertEquals(ApiClientStatus.ACTIVE, apiClient.snapshot().status());
    }

    @Test
    void mustBeEqualWhenIdsMatch() {
        UUID id = UUID.randomUUID();

        ApiClient apiClientA = ApiClient.newApiClient(id, UUID.randomUUID(), "ERP client", null);
        ApiClient apiClientB = ApiClient.newApiClient(id, UUID.randomUUID(), "CRM client", "CRM");
        ApiClient apiClientC = ApiClient.newApiClient(UUID.randomUUID(), UUID.randomUUID(), "Integration client", "Integration client");

        Assertions.assertAll(
            () -> assertEquals(apiClientA, apiClientB),
            () -> assertNotEquals(apiClientA, apiClientC)
        );
    }
}

File: services/identity-service/src/test/kotlin/br/com/deltaglobalbank/identity/domain/user/ApiClientTests.kt
package br.com.deltaglobalbank.identity.domain.user

import br.com.deltaglobalbank.identity.domain.apiClient.ApiClient
import br.com.deltaglobalbank.identity.domain.apiClient.ApiClientStatus
import io.mockk.junit5.MockKExtension
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.assertAll
import org.junit.jupiter.api.extension.ExtendWith
import java.util.UUID
import kotlin.test.Test
import kotlin.test.assertNotEquals

@ExtendWith(MockKExtension::class)
class ApiClientTests {
    @Test
    fun `must initialize with active status`() {
        val apiClient = ApiClient.newApiClient(
            id = UUID.randomUUID(),
            tenantId = UUID.randomUUID(),
            name = "client",
            description = null
        )
        assertEquals(ApiClientStatus.ACTIVE, apiClient.snapshot().status)
    }

    @Test
    fun `must be equal when ids match`() {
        val id = UUID.randomUUID()

        val ApiClientA = ApiClient.newApiClient(id = id, tenantId = UUID.randomUUID(), name = "ERP client", description = null)
        val ApiClientB = ApiClient.newApiClient(id = id, tenantId = UUID.randomUUID(), name = "CRM client", description = "CRM")
        val ApiClientC = ApiClient.newApiClient(id = UUID.randomUUID(), tenantId = UUID.randomUUID(), name = "Integration client", description = "Integration client")

        assertAll(
            {assertEquals(ApiClientA, ApiClientB)},
            { assertNotEquals(ApiClientA, ApiClientC) }
        )

    }
}